docs(apisix): wrote transformation/transcode-http-to-grpc (#141)
api7/docs: apisix_versioned_docs/version-3.2.0/how-to-guide/transform/grpc.md

---
title: Transcode HTTP to gRPC
slug: /how-to-guide/transform/grpc
---

> [gRPC](https://grpc.io/) is a open source high performance Remote Procedure Call (RPC) framework based on HTTP/2 protocol. It uses [protocol buffers](https://protobuf.dev/) (a.k.a. _protobuf_) as the interface description language. gRPC framework provides a portable mechanism for load balancing, link tracing, health checking, and authentication, effectively connecting multiple services between datacenters.

Reviewer: ```suggestion
[gRPC](https://grpc.io/) is an open-source high performance Remote Procedure Call (RPC) framework based on HTTP/2 protocol. It uses [protocol buffers](https://protobuf.dev/) (a.k.a. _protobuf_) as the interface description language. gRPC framework provides a portable mechanism for load balancing, link tracing, health checking, and authentication, effectively connecting multiple services between datacenters.
```

---
title: gRPC Proxy
slug: /how-to-guide/transform/grpc
---

[gRPC](https://grpc.io/) is a open source high performance Remote Procedure Call (RPC) framework based on HTTP/2 protocol. It uses [protocol buffers](https://protobuf.dev/) (a.k.a. _protobuf_) as the interface description language. The gRPC service provides portable mechanism for load balancing, link tracing, health checking, and authentication, effectively connecting multiple services between datacenters.

> APISIX supports proxy gRPC using the plugin `grpc-transcode`, which invokes gRPC services in a RESTful way. The plugin supports specifying the contents of `.proto` files in Apache APISIX and implementing proxies for different gRPC services through user-defined gRPC services.

Reviewer: ```suggestion
APISIX supports transcoding between HTTP and gRPC using the plugin `grpc-transcode`. The plugin supports specifying the contents of `.proto` files in Apache APISIX and implementing proxies for different gRPC services through user-defined gRPC services.
```
